src/event.js

```diff
@@ -554,7 +554,9 @@ function leverageNative( el, type, expectSync ) {
 			if ( ( event.isTrigger & 1 ) && this[ type ] ) {
 
 				// Interrupt processing of the outer synthetic .trigger()ed event
-				if ( !saved ) {
+				// Saved data should be false in such cases, but might be a leftover capture object
+				// from an async native handler (gh-4350)
+				if ( !saved.length ) {
```

Comment:
This `if` branch can be executed if `length` is falsy which might suggest it may happen for the `arguments` saved below... but there will always be at least one argument passed as the event argument is always passed.

It's a little fragile but we have a test for that so I guess we're fine.

Reply:
Added source comments to explain.